File: src/test/java/net/bramp/ffmpeg/builder/MetadataSpecTest.java
package net.bramp.ffmpeg.builder;

import static net.bramp.ffmpeg.builder.MetadataSpecifier.*;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

import org.junit.Test;

public class MetadataSpecTest {

  @Test
  public void testMetaSpec() {
    assertThat(MetadataSpecifier.Companion.global().getSpec(), is("g"));
    assertThat(MetadataSpecifier.Companion.chapter(1).getSpec(), is("c:1"));
    assertThat(MetadataSpecifier.Companion.program(1).getSpec(), is("p:1"));
    assertThat(MetadataSpecifier.Companion.stream(1).getSpec(), is("s:1"));
    assertThat(MetadataSpecifier.Companion.stream(StreamSpecifier.Companion.id(1)).getSpec(), is("s:i:1"));
  }
}

File: src/test/kotlin/net/bramp/ffmpeg/kotlin/builder/MetadataSpecTest.kt
package net.bramp.ffmpeg.kotlin.builder

import io.kotest.matchers.shouldBe
import net.bramp.ffmpeg.kotlin.builder.MetadataSpecifier.Companion.chapter
import net.bramp.ffmpeg.kotlin.builder.MetadataSpecifier.Companion.global
import net.bramp.ffmpeg.kotlin.builder.MetadataSpecifier.Companion.program
import net.bramp.ffmpeg.kotlin.builder.MetadataSpecifier.Companion.stream
import net.bramp.ffmpeg.kotlin.builder.StreamSpecifier.Companion.id
import org.junit.Test

class MetadataSpecTest {

  @Test
  fun testMetaSpec() {
    global().spec shouldBe "g"
    chapter(1).spec shouldBe "c:1"
    program(1).spec shouldBe "p:1"
    stream(1).spec shouldBe "s:1"
    stream(id(1)).spec shouldBe "s:i:1"
  }
}
